lib/chef/provider/windows_task.rb

```diff
@@ -243,6 +242,37 @@ def task_need_update?
         false
       end
 
+      # Comparing random_delay values using ISO8601::Duration object Ref: https://github.com/arnau/ISO8601/blob/master/lib/iso8601/duration.rb#L18-L23
+      # di = ISO8601::Duration.new(65707200)
+      # ds = ISO8601::Duration.new('P65707200S')
+      # dp = ISO8601::Duration.new('P2Y1MT2H')
+      # di == dp # => true
+      # di == ds # => true
+      def random_delay_updated?
+        if new_resource.random_delay.nil?
+          false
+        elsif current_resource.random_delay.nil? && new_resource.random_delay == "PT0S" # when user sets random_dealy to 0 sec
+          false
+        elsif current_resource.random_delay.nil? && !new_resource.random_delay.nil?
```

Comment:
You've already returned if new_resource.random_delay.nil? above so at this point you know new_resource.random_delay is not nil and shouldn't need to check it again